File: Gemini 1.5 Pro/Python/2-5.py
class BankAccount:
    def __init__(self, account_number, initial_balance=0):
        self.account_number = account_number
        self.balance = initial_balance

    def deposit(self, amount):
        if amount > 0:
            self.balance += amount
            print(f"Deposited ${amount:.2f}. New balance: ${self.balance:.2f}")
        else:
            print("Invalid deposit amount.")

    def withdraw(self, amount):
        if amount > 0 and amount <= self.balance:
            self.balance -= amount
            print(f"Withdrew ${amount:.2f}. New balance: ${self.balance:.2f}")
        elif amount <= 0:
            print("Invalid withdrawal amount.")
        else:
            print("Insufficient funds.")

    def check_balance(self):
        print(f"Account balance: ${self.balance:.2f}")


# Example usage:
account = BankAccount("1234567890", 1000)  # Create an account with initial balance

account.deposit(500)
account.withdraw(200)
account.check_balance()

File: Gemini 1.5 Pro/C++/2-5.c
#include <iostream>

using namespace std;

class BankAccount {
private:
  double balance;

public:
  BankAccount(double initialBalance = 0.0) {
    balance = initialBalance;
  }

  void deposit(double amount) {
    balance += amount;
    cout << "Deposit successful. New balance: " << balance << endl;
  }

  void withdraw(double amount) {
    if (balance >= amount) {
      balance -= amount;
      cout << "Withdrawal successful. New balance: " << balance << endl;
    } else {
      cout << "Insufficient balance." << endl;
    }
  }

  void checkBalance() {
    cout << "Current balance: " << balance << endl;
  }
};

int main() {
  BankAccount account(1000.0); // Initial balance of 1000.0

  int choice;
  double amount;

  do {
    cout << "\nBank Account Menu:" << endl;
    cout << "1. Deposit" << endl;
    cout << "2. Withdraw" << endl;
    cout << "3. Check Balance" << endl;
    cout << "4. Exit" << endl;
    cout << "Enter your choice: ";
    cin >> choice;

    switch (choice) {
      case 1:
        cout << "Enter amount to deposit: ";
        cin >> amount;
        account.deposit(amount);
        break;
      case 2:
        cout << "Enter amount to withdraw: ";
        cin >> amount;
        account.withdraw(amount);
        break;
      case 3:
        account.checkBalance();
        break;
      case 4:
        cout << "Exiting..." << endl;
        break;
      default:
        cout << "Invalid choice." << endl;
    }
  } while (choice != 4);

  return 0;
}

